language/stdlib/modules/libra_coin.move

```diff
@@ -15,20 +15,33 @@ module LibraCoin {
     resource struct MarketCap {
         // The sum of the values of all LibraCoin::T resources in the system
         total_value: u128,
+        // A holding area where funds that will subsequently be burned wait while their underyling
+        // assets are sold off-chain
+        preburn: T,
     }
 
-    // Return a reference to the MintCapability published under the sender's account. Fails if the
-    // sender does not have a MintCapability.
-    // Since only the Association account has a mint capability, this will only succeed if it is
-    // invoked by a transaction sent by that account.
-    public fun mint_with_default_capability(amount: u64): T acquires MintCapability, MarketCap {
+    // Return `amount` coins created using the MintCapability published under the sender's account.
+    // Fails if the sender does not have a published MintCapability.
+    public fun mint_with_capability(amount: u64): T acquires MintCapability, MarketCap {
         mint(amount, borrow_global<MintCapability>(Transaction::sender()))
     }
 
+    // Move `coin` to the preburn holding area using the MintCapability published under the sender'
+    // account. Fails if the sender does not have a published MintCapability.
+    public fun preburn_with_capability(coin: T) acquires MintCapability, MarketCap {
```

Comment:
Ditto for the "preburn" and "burn" functions here.... I don't think they should have "_with_capability" in their names.